### PR TITLE
fix issue `notifyPropertyChange` not working for plain obj

### PR DESCRIPTION
**Pull request by bekzod**
possible fix for #16427

### CODE REVIEW

**Review by rwjblue — COMMENTED** [review]
IMHO, the fix needs to be in `descriptorFor` itself otherwise we will continue to find other APIs that don’t work correctly.

File: packages/ember-metal/lib/property_events.js [hunk 1]
```diff
@@ -86,8 +86,7 @@ function notifyPropertyChange(obj, keyName, _meta) {
 
   let possibleDesc = descriptorFor(obj, keyName, meta);
 
-  // shouldn't this mean that we're watching this key?
-  if (possibleDesc !== undefined && possibleDesc.didChange) {
+  if (possibleDesc !== undefined && 'didChange' in possibleDesc) {
```
Comment (krisselden) [inline packages/ember-metal/lib/property_events.js]: this change seems unrelated and I'm curious why?

File: packages/ember-metal/lib/meta.js [hunk 2]
```diff
@@ -693,7 +693,17 @@ export function descriptorFor(obj, keyName, _meta) {
   let meta = _meta === undefined ? peekMeta(obj) : _meta;
 
   if (meta !== undefined) {
-    return meta.peekDescriptors(keyName);
+    let descriptor = meta.peekDescriptors(keyName);
+
+    if (descriptor === undefined && keyName in obj) {
+      // #GH16427 in case defined without `defineProperty`
+      let possibleDesc = obj[keyName];
+      if (isDescriptor(possibleDesc) && 'didChange' in possibleDesc) {
```
Comment (krisselden) [inline packages/ember-metal/lib/meta.js]: 'didChange' in possibleDesc doesn't belong in descriptorFor

Also we would need to move this deprecation and fallback code and make sure we don't have the fallback code elsewhere.
https://github.com/emberjs/ember.js/blob/470dddb9bbea32f0bf71219740b22c8dc8ecf68d/packages/ember-metal/lib/property_get.js#L117

**Review by krisselden — CHANGES_REQUESTED** [review]

File: packages/ember-metal/lib/property_events.js (hunk 1 above)
Comment (bekzod) [reply packages/ember-metal/lib/property_events.js]: just to avoid `boolean coercion`

Comment (krisselden) [reply packages/ember-metal/lib/property_events.js]: Last time I checked `in` was worse than boolean coercion in v8 (cc @bmeurer ) it is better to do possibleDesc.didChange !== undefined

Comment (bmeurer) [reply packages/ember-metal/lib/property_events.js]: I'd go with `possibleDesc.didChange !== undefined`. The `in` operator is not backed by ICs so it only pays off if the objects have vastly different shapes all the time.

Comment (krisselden) [reply packages/ember-metal/lib/property_events.js]: Thanks for the quick response

**Comment by krisselden** [comment]
superseded by #16653
